### PR TITLE
Extract zoom commit messages as a separate component

### PR DESCRIPTION
**Pull request by gerhean**
Fixes [#1340]
```
v-zoom currently uses window.toggleNext() to show and
hide commit messages.

window.toggleNext() directly modifies html (add classes and
elements directly), and so might conflict with how Vue works.

As a result, using window.toggleNext() creates code which is
hard to debug and maintain.

Lets rewrite v-zoom to use the Vue way of hiding and showing
elements by using v-if.

In order to do this, the code which handles the display of zoom
commit messages has been moved into a separate file. This also
helps with code readability.

Let's remove window.toggleNext().
```

### CODE REVIEW

**Comment by fzdy1914** [comment]
@gerhean Is this part of #1412?

**Comment by gerhean** [comment]
no

**Comment by damithc** [comment]
The width of the commit message body seems narrower than necessary? Or is my eye fooling me?

**Comment by gerhean** [comment]
I can't see any difference in width, is it possible to provide a screenshot?

**Comment by damithc** [comment]
Compare these two:
image
image

Yours seems noticeably narrower. Perhaps following a wrong width limit?

**Comment by HCY123902** [comment]
Can you change the commit message to make it a bit wider according to the suggestion given? Thanks

**Comment by gerhean** [comment]
I'm not sure if I'm blind but I literally can't see any difference...
image
image

**Comment by damithc** [comment]
> I'm not sure if I'm blind but I literally can't see any difference...

I think @HCY123902 was referring to https://github.com/reposense/RepoSense/pull/1462#issuecomment-792237869
Your propose commit message is line wrapping too early.

**Comment by gerhean** [comment]
Ok nevermind I think I might actually be blind, I'll be fixing it right away.

**Comment by 0blivious** [comment]
I agree with the idea of extracting out the component. However, this approach seems to introduce some even more complicated communication between zoom and the message body component. Why don't you follow the approach we used in the authorship panel, which uses v-show to control the toggle of the individual elements?

**Comment by gerhean** [comment]
@0blivious I don't really want to introduce a new property into the `slice` object, since it is already hard to keep track of what properties are in what object. Plus it means I have to shallow copy the `slice` object as well.

I don't believe that the communication process is particularly complicated, given that I only added about 4 lines into v_zoom.js, while most of the communication logic is in v_zoom_commit.js, which I believe is a reasonable abstraction.

Following the approach we used in the authorship panel would increase the amount of code in v_zoom.js, making it overall harder to read.

File: frontend/src/static/js/v_zoom_commit.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  created() {
+    if (this.defaultExpansionState) {
+      this.toggleExpansion();
+    }
+  },
+  beforeDestroy() {
+    if (this.isExpanded) {
+      this.$emit('decrement-expanded-count');
+    }
+  },
```
Comment (0blivious) [inline frontend/src/static/js/v_zoom_commit.js:70]: May I know that is the case for this beforeDestroy?

Comment (0blivious) [inline frontend/src/static/js/v_zoom_commit.js:70]: May I know that is the case for this beforeDestroy?

File: frontend/src/static/js/v_zoom.js [hunk 2]
```diff
@@ -15,7 +15,12 @@ function initialState() {
 window.vZoom = {
   template: window.$('v_zoom').innerHTML,
   data() {
-    return initialState();
+    return {
+      defaultExpansionStateSignal: false,
```
Comment (0blivious) [inline frontend/src/static/js/v_zoom.js:19]: Can we have some other naming like toExpandAll?

File: frontend/src/static/js/v_zoom_commit.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/* global Vuex */
+
+window.vZoomCommit = {
+  props: [
+      'defaultExpansionState',
+      'defaultExpansionStateSignal',
```
Comment (0blivious) [inline frontend/src/static/js/v_zoom_commit.js:6]: Do we really need both state and signal? Is it possible to just use one?

**Pull request closed without merging**